Settings.c

```diff
@@ -73,20 +73,27 @@ static void writeQuotedList(FILE* fp, char** list) {
 
 */
 
-void Settings_delete(Settings* this) {
-   free(this->filename);
-   free(this->initialFilename);
-   for (unsigned int i = 0; i < HeaderLayout_getColumns(this->hLayout); i++) {
+static void Settings_deleteColumns(Settings* this) {
+   for (size_t i = 0; i < HeaderLayout_getColumns(this->hLayout); i++) {
       String_freeArray(this->hColumns[i].names);
       free(this->hColumns[i].modes);
    }
    free(this->hColumns);
+}
+
+static void Settings_deleteScreens(Settings* this) {
    if (this->screens) {
-      for (unsigned int i = 0; this->screens[i]; i++) {
+      for (size_t i = 0; this->screens[i]; i++)
          ScreenSettings_delete(this->screens[i]);
-      }
       free(this->screens);
    }
+}
+
+void Settings_delete(Settings* this) {
+   free(this->filename);
+   free(this->initialFilename);
+   Settings_deleteColumns(this);
+   Settings_deleteScreens(this);
```

Comment:
```suggestion
   Settings_deleteColumns(this);
   Settings_deleteScreens(this);
   free(this->filename);
   free(this->initialFilename);
```

Remove sub-structures first. If (I know that's not the case here) the cleanup routines needed any access on the main structure, this would provide a more complete environment, thus reducing the dependencies for what may not be accessed.

Reply:
Since this was a simple cleanup, I preferred not to change the logic or ordering at all.